AndroidTest/app/src/main/java/cloudant/com/androidtest/keyprovider/AndroidKeyProviderTests.java

```diff
@@ -0,0 +1,173 @@
+package cloudant.com.androidtest.keyprovider;
+
+import android.test.AndroidTestCase;
+
+import com.cloudant.sync.datastore.encryption.AndroidKeyProvider;
+import com.cloudant.sync.datastore.encryption.DPKException;
+import com.cloudant.sync.datastore.encryption.EncryptionKey;
+import com.cloudant.sync.datastore.encryption.KeyManager;
+import com.cloudant.sync.datastore.encryption.KeyStorage;
+
+import java.util.Arrays;
+
+public class AndroidKeyProviderTests extends AndroidTestCase {
+
+    public void testCreateProviderWithIdentifier() {
+        AndroidKeyProvider provider = new AndroidKeyProvider(getContext(), "provider1password",
+                ProviderTestUtil.getUniqueIdentifier());
+
+        EncryptionKey createdKey = provider.getEncryptionKey();
+        assertNotNull("EncryptionKey was not created", createdKey);
+
+        EncryptionKey loadedKey = provider.getEncryptionKey();
+        assertNotNull("EncryptionKey was not loaded", loadedKey);
+
+        assertTrue("loadedKey did not match createdKey", Arrays.equals(createdKey.getKey(),
+                loadedKey.getKey()));
+    }
+
+    public void testCreateProviderWithManager() {
+        KeyStorage storage = new KeyStorage(getContext(), ProviderTestUtil.getUniqueIdentifier());
+        KeyManager manager = new KeyManager(storage);
+        AndroidKeyProvider provider = new AndroidKeyProvider("provider1password", manager);
+
+        EncryptionKey createdKey = provider.getEncryptionKey();
+        assertNotNull("EncryptionKey was not created", createdKey);
+
+        EncryptionKey loadedKey = provider.getEncryptionKey();
+        assertNotNull("EncryptionKey was not loaded", loadedKey);
+
+        assertTrue("loadedKey did not match createdKey", Arrays.equals(createdKey.getKey(),
+                loadedKey.getKey()));
```

Comment:
I think It also necessary to test that `AndroidKeyProvider.getEncryptionKey` does not call `KeyManager. generateAndSaveKeyProtectedByPassword` if there is a key already stored.